sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -4202,6 +4202,19 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       }
     }
   }
+
+  test("SPARK-36352: Optimizer should keep schema name not changed") {
+    withTable("t") {
+      sql("CREATE TABLE t1(a INT, b INT) USING PARQUET")
+      sql(
+        """
+          |INSERT INTO t1 VALUES
+          |(1, 2),
+          |(2, 3)""".stripMargin)
+      val df = sql("SELECT C FROM (SELECT a + b AS c FROM t1)")
```

Comment:
This was broken before due to the issue in `ProjectionOverSchema`?

Reply:
Removed